File: src/main/java/eu/printingin3d/javascad/models/surfaces/bicubic/Vertex.java
package eu.printingin3d.javascad.models.surfaces.bicubic;

import eu.printingin3d.javascad.coords.V3d;

public class Vertex {
   /**
    * Vertex position.
    */
   public final V3d position;
   /**
    * Vertex normal.
    */
   public V3d normal = new V3d(0,0,0);

   public Vertex(V3d position, V3d normal) {
      this.position = position;
      this.normal = normal;
   }

   public Vertex(V3d position) {
      this.position = position;
   }
}

File: javascad/src/main/kotlin/eu/printingin3d/javascad/manifold/Vertex.kt
package eu.printingin3d.javascad.manifold

import eu.printingin3d.javascad.coords.V3d

class Vertex(val id: Long, var position: V3d) {
    var outgoing: HalfEdge? = null
    var normal: V3d = V3d.ZERO

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        return other is Vertex && id == other.id
    }

    override fun hashCode(): Int = id.hashCode()
}
